src/player.cpp

```diff
@@ -7456,6 +7456,58 @@ void player::rooted()
     }
 }
 
+bool player::add_faction_warning( const faction_id id )
+{
+    std::map<faction_id, std::pair<int, time_point>>::iterator it;
+    it = warning_record.find( id );
```

Comment:
```suggestion
    const auto it = warning_record.find( id );
```
And remove the declaration of `it` above.
And if actually set this variable, why are you not using it in the code below instead of finding the element again, and again, and again, and again and so on?